### PR TITLE
SMTP e-mail settings on a store level 

### LINKED ISSUE #468: SMTP e-mail settings on a store level
We should we allow non-admins to configure SMTP setting, possibly on store-level.

Use case: 

1. The user wants to have a payment button e-mail notification, but uses someone else’s server, sees the e-mail settings but can’t create SMPT since he’s not the admin.

2. Same goes for #313 I assume one would want to send e-mail confirmations or a thank you emails to whoever supports the campaign.

### PR DESCRIPTION
**Pull request by Kukks**
closes #468

### CODE REVIEW

**Comment by NicolasDorier** [comment]
The email sender should not know about `StoreId`. Instead, the email sender instance should be created from the store instead.

**Comment by Kukks** [comment]
@NicolasDorier  Should I make an `EmailSenderFactory` then?  I'm trying to recycle the instances so that we dont have to recreate them for each background task

**Comment by NicolasDorier** [comment]
Yes I think having a EmailSenderFactory is good.

**Comment by Kukks** [comment]
@NicolasDorier  done :)

**Comment by NicolasDorier** [comment]
No `ContinueWith` please, just call it without waiting.
To remove the warning, I think you can use `Task.Run(blablaAsync())`

File: BTCPayServer/Services/Mails/EmailSenderFactory.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+        public EmailSenderFactory(IBackgroundJobClient jobClient,
+            SettingsRepository repository,
+            StoreRepository storeRepository,
+            EventAggregator eventAggregator)
+        {
+            _JobClient = jobClient;
+            _Repository = repository;
+            _StoreRepository = storeRepository;
+            eventAggregator.Subscribe<EmailSettingsUpdated>(updated =>
+                _CachedClients.TryRemove(updated.StoreId ?? ServerSettingsKey, out var x));
```
Comment (NicolasDorier) [inline BTCPayServer/Services/Mails/EmailSenderFactory.cs]: you never unsubscribe, this is a leak.

**Comment by Kukks** [comment]
```

> No `ContinueWith` please, just call it without waiting.
> To remove the warning, I think you can use `Task.Run(blablaAsync())`

Already fixed it

**Comment by Kukks** [comment]
this is ready for review again @NicolasDorier

**Comment by NicolasDorier** [comment]
I will rewrite this. `GetEmailSender` does not have to be asking. One sec.

**Pull request merged**